File: src/main/java/com/example/CargoTracking/dto/DomesticPerformance.java
package com.example.CargoTracking.dto;

import lombok.*;

import java.time.LocalDateTime;
import java.time.LocalTime;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Builder
public class DomesticPerformance {
    private Long id;
    private String preAlertNumber;
    private String referenceNumber;
    private String origin;
    private String destination;
    private String route;
    private String vehicle;
    private Integer shipments; // no of shipments
    private Integer pallets; // no of pallets
    private String occupancy;
    private Integer bags;
    private LocalTime planedEtd;
    private LocalTime planedEta;
    private LocalDateTime atd;
    private LocalDateTime ata;
    private Long planedEtdVsAtd;
    private Long planedEtaVsAta;
    private Long transitTime;
}

File: frontend/src/app/model/DomesticPerformance.ts
export interface DomesticPerformance{
    id:number;
    preAlertNumber:string|null;
    referenceNumber:string|null;
    origin:string|null;
    destination:string|null;
    route:string|null;
    vehicle:string|null;
    shipments:number|null; // no of shipments
    pallets:number|null; // no of pallets
    occupancy:string|null;
    bags:number|null;
    planedEtd:Date|string|null;
    planedEta:Date|string|null;
    atd:Date|string|null;
    ata:Date|string|null;
    planedEtdVsAtd:number|null;
    planedEtaVsAta:number|null;
    transitTime:number|null;
}